chapters/data-processing.tex

```diff
@@ -0,0 +1,725 @@
+Ensuring that observations are uniquely and fully identified
+is arguably the most important step in data cleaning.
+It may be the case that the variables expected to be unique identifiers are either incomplete or contain duplicates.
+It is also possible for a dataset to have no unique identifier, 
+or that the identifier involves a long string that is difficult to work with, such as a full name.
+In such cases, cleaning begins by carefully creating a variable that uniquely identifies the data.
+Note that while modern survey tools create unique identifiers for each data submission,
+that is not the same as having a unique ID variable for each observation in the sample,
+as there can be multiple submissions for the same observation.
+The unique identifier will be used to link data on observations in one dataset 
+with data on the same observations in other datasets, 
+such as other rounds of data collection and different data sources.
+Therefore, you must keep a canonical dataset of all observations in your data.
+This \textbf{master dataset}\sidenote{\url{
+		https://dimewiki.worldbank.org/Master\_Data\_Set}}
+is discussed in detail in Chapter 3, 
+and contains a record of all units ever encountered for each relevant level of observation,
+as well as their corresponding unique identifiers.
+
+
+% Merchand
+Checking for duplicated entries and making sure that each observation has a unique identifier are usually part of data quality monitoring.
+The lack of a unique identifier should be addressed as soon as data is received,
+so you can keep track of observations.
+\texttt{ieduplicates} and \texttt{iecompdup},
+two Stata commands included in the \texttt{iefieldkit}
+package\index{iefieldkit},\sidenote{\url{https://dimewiki.worldbank.org/iefieldkit}}
+create an automated workflow to identify, correct and document
+occurrences of duplicate entries.
+One advantage of using \texttt{ieduplicates}\sidenote{\url{
+		https://dimewiki.worldbank.org/wiki/ieduplicates}} 
+to correct duplicated entries is that it creates \textit{duplicates reports}
+that record which corrections made and why.
+Even if you are not using this command,
+it is important to include similar records in the \texttt{Documentation} folder.
+
+\subsection{Tidying raw data}
+
+Though raw data can be acquired in all shapes and sizes,
+development data is most commonly transferred as one or multiple tables.\sidenote{\textbf{Table:}
+	a two-dimensional dataset formed by rows and columns. 
+	Also called ``rectangular data''.
+	Examples of non-rectangular data are a text, or files, such as images.}
+Similarly, these tables can organize information in multiple ways,
+and not all of them result in easy-to-handle datasets.
+Fortunately, a vast literature of database management has identified the format
+that makes interacting with the data as easy as it can be.
+When applied to statistics, we call data in such format \textbf{tidy}.\sidenote{\textbf{Tidy data:}
```

Comment:
```suggestion
In database management this is called \textit{normalization},
but when applied to statistics, we call data in such format \textbf{tidy}.\sidenote{\textbf{Tidy data:}
```
Since you are mentioning database management literature I think we should give the term so people can look it up if they want. This feels more substantial and more credibly stands on the shoulders of database giants.

Reply:
cool, I didn't know that term